verilog/tools/formatter/verilog_format.cc

```diff
@@ -216,22 +194,62 @@ int main(int argc, char** argv) {
     }
     if (absl::GetFlag(FLAGS_failsafe_success)) {
       // original text was preserved, and --inplace modification is skipped.
-      return 0;
+      return true;
     }
-    return 1;
+    return false;
   }
 
   // Safe to write out result, having passed above verification.
   if (inplace && !is_stdin) {
-    if (!verible::file::SetContents(filename, formatted_output)) {
-      std::cerr << "Error writing to file: " << filename << std::endl;
-      std::cerr << "Printing to stdout instead." << std::endl;
-      std::cout << formatted_output;
-      return 1;
+    // Don't write if the output is exactly as the input, so that we don't mess
+    // with tools that look for timestamp changes (such as make).
+    if (content != formatted_output) {
+      if (!verible::file::SetContents(filename, formatted_output)) {
+        std::cerr << "Error writing to file: " << filename << std::endl;
+        std::cerr << "Printing to stdout instead." << std::endl;
+        std::cout << formatted_output;
+        return false;
+      }
     }
   } else {
     std::cout << formatted_output;
   }
 
-  return 0;
+  return true;
+}
+
+int main(int argc, char** argv) {
+  const auto usage = absl::StrCat("usage: ", argv[0],
+                                  " [options] <file>\n"
+                                  "To pipe from stdin, use '-' as <file>.");
+  const auto file_args = verible::InitCommandLine(usage, &argc, &argv);
+
+  if (file_args.size() == 1) {
+    std::cerr << absl::ProgramUsageMessage() << std::endl;
+    // TODO(hzeller): how can we append the output of --help here ?
+    return 1;
+  }
+
+  // Parse LineRanges into a line set, to validate the --lines flag(s)
+  verilog::formatter::LineNumberSet lines_to_format;
+  if (!verible::ParseInclusiveRanges(
+          &lines_to_format, LineRanges::values.begin(),
+          LineRanges::values.end(), &std::cerr, '-')) {
+    std::cerr << "Error parsing --lines." << std::endl;
+    std::cerr << "Got: --lines=" << AbslUnparseFlag(LineRanges()) << std::endl;
+    return 1;
+  }
+
+  if (!lines_to_format.empty() && file_args.size() > 2) {
+    std::cerr << "Note, applying --lines rules to all files" << std::endl;
```

Comment:
This feels weird/surprising to do, can we error out instead?  Maybe see what clang_format does in this case.

Reply:
Done.